### PR TITLE
Change pandas to never let objects share `_data`, create views instead

### PR DESCRIPTION
**Pull request by nickeubank**
Closes Issue #11814 

Several pandas operations create objects that share `_data` objects. In particular, all of the following result in `df_new`s and `df` sharing the same `_data` (i.e. `df_new._data is df._data` returns `True`). 

```
df = DataFrame({'col1':[0,1,2], 'col2':[2,3,4]})
df_new = DataFrame(df)
df_new1 = df.loc[:,:]
df_new2 = df.iloc[:,:]
```

Needed for PR #11500

### CODE REVIEW

File: pandas/core/frame.py [hunk 1]
```diff
@@ -217,6 +217,7 @@ def __init__(self, data=None, index=None, columns=None, dtype=None,
             dtype = self._validate_dtype(dtype)
 
         if isinstance(data, DataFrame):
+            data = data.iloc[:,:]
             data = data._data
```
Comment (jreback) [inline pandas/core/frame.py]: you need to define ``BlockManager.view()`` this will do all of the heavy lifting, everything else will flow from there

**Comment by jreback** [comment]
xref #11970 

this idea will be incorporated there.

If you want to update before then (as this might take a while), feel free to reopen.

File: pandas/core/frame.py (hunk 1 above)
Comment (nickeubank) [inline pandas/core/frame.py]: @jreback OK -- best way to get a view at the level of BlockManager object? I understand how to get them at the level of DataFrames (`df.iloc[:,:]` works safely, but I know it's a little hacky) , but don't know how best to get a view reliably.

**Comment by nickeubank** [comment]
gonna try, thanks

**Pull request closed without merging**